src/appstore-reviews.ts

```diff
@@ -0,0 +1,235 @@
+import { Data, AppStoreConfig, PublishedReviews } from "./global-types"
+import axios from "axios"
+import allRegions from './regions.json'
+
+export default class AppStoreReviews {
+
+    BASE_URL = "https://itunes.apple.com"
+    STORE_NAME = "App Store"
+
+    async fetch(config: AppStoreConfig, publishedReviews: PublishedReviews): Promise<Data> {
+        let { id, regions } = config
+
+        const appInformation = await this.fetchAppInformation(id)
+
+        if (regions == 'all') {
+            regions = allRegions
+        }
+
+        if (regions.length == 0) {
+            throw "At least one region should be define in configuration"
+        }
+
+        const requests = (regions as string[]).map(region => this.fetchAppStoreReviews(id, config.pageRange || 5, region))
+        const result = await Promise.all(requests)
+
+        let reviews: Review[] = []
+        var newReviews = reviews.concat(...result).filter((review) => {
+            if (publishedReviews[id] !== undefined && publishedReviews[id].indexOf(review.id) >= 0) {
+                return false
+            }
+            return true
+        })
+
+        var newReviewsMap: PublishedReviews = {}
+        newReviewsMap[id] = newReviews.map(this.mapReviewId)
+
+        return {
+            messages: newReviews.map(review => {
+                if (config.generateMessageFromReview !== undefined) {
+                    return config.generateMessageFromReview.call(this, review, appInformation, config)
+                } else {
```

Comment:
This is more code style but removing else would make code more readable imho.

Reply:
fixed